Kickstarter-iOS/Features/FloatingTabBar/FloatingTabBar.swift

```diff
@@ -0,0 +1,208 @@
+import KDS
+import UIKit
+
+/* A custom floating-style `UITabBar` that visually groups tab items inside a
+ - Sets tabs in a centered container using a horizontal `UIStackView`.
+ - Draws a background “pill” behind the tab items, including shadow.
+ - Animates a selection indicator view behind the currently selected tab
+ - Hides tab item titles and presents an icon-only navigation experience.
+ */
+
+final class FloatingTabBar: UITabBar {
+  private enum Constants {
+    static let tabBarItemSize: CGFloat = 40
+    static let tabBarWidth: CGFloat = 152
+    static let tabBarCornerRadius: CGFloat = 16
+    static let tabBarShadowOpacity: Float = 0.28
+    static let tabBarShadowRadius: CGFloat = 28
+    static let tabBarShadowOffsetY: CGFloat = 8
+    static let tabBarVerticalPadding: CGFloat = 8
+    static let tabBarHorizontalInset: CGFloat = 12
+
+    static let selectedTabBackgroundCornerRadius: CGFloat = 8
+    static let selectedTabBackgroundSize: CGSize = CGSize(
+      width: Constants.tabBarItemSize,
+      height: Constants.tabBarItemSize
+    )
+    static let selectedTabAnimationDuration: TimeInterval = 0.18
+  }
+
+  private let tabBarBackgroundView = UIView()
+  private let selectedTabBackgroundView = UIView()
+
+  private let tabsStackView: UIStackView = {
+    let view = UIStackView()
+    view.axis = .horizontal
+    view.alignment = .center
+    view.distribution = .equalSpacing
+    view.isUserInteractionEnabled = false
+    return view
+  }()
+
+  // MARK: - Init
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+
+    self.backgroundImage = UIImage()
+    self.shadowImage = UIImage()
+    self.isTranslucent = true
+    self.backgroundColor = .clear
+
+    self.tintColor = Colors.FloatingTabBar.iconColorSelected.uiColor()
+    self.unselectedItemTintColor = Colors.FloatingTabBar.iconColorUnselected.uiColor()
+
+    let appearance = UITabBarAppearance()
+    appearance.configureWithTransparentBackground()
+    appearance.stackedLayoutAppearance.selected.iconColor = Colors.FloatingTabBar.iconColorSelected.uiColor()
+    appearance.stackedLayoutAppearance.normal.iconColor = Colors.FloatingTabBar.iconColorUnselected.uiColor()
+
+    // Hide icon labels.
+    appearance.stackedLayoutAppearance.normal.titleTextAttributes = [
+      .foregroundColor: UIColor.clear
+    ]
+    appearance.stackedLayoutAppearance.selected.titleTextAttributes = [
+      .foregroundColor: UIColor.clear
+    ]
+
+    self.standardAppearance = appearance
+    self.scrollEdgeAppearance = appearance
+
+    self.setupSubviews()
+  }
+
+  required init?(coder: NSCoder) {
+    super.init(coder: coder)
```

Comment:
Nit: This should be a `fatalError` since nobody should be using this initializer.
```suggestion
    fatalError("init(coder:) has not been implemented")
```